libs/queue/src/main.sw

```diff
@@ -0,0 +1,98 @@
+// The `Queue` type corresponds to the same data structure.
+library;
+
+use std::vec::Vec;
+
+pub struct Queue<T> {
+    vec: Vec<T>,
+}
+
+impl<T> Queue<T> {
+    /// Create a new `Queue`
+    pub fn new() -> Self {
+        Self {
+            vec: Vec::new(),
+        }
+    }
+
+    /// Checks a `Queue` for emptiness
+    pub fn is_empty(self) -> bool {
+        self.vec.is_empty()
+    }
+
+    /// Gets the number of elements in the `Queue`
+    pub fn len(self) -> u64 {
+        self.vec.len()
+    }
+
+    /// Enqueues a `Queue`
+    pub fn enqueue(ref mut self, item: T) {
+        self.vec.push(item);
+    }
+
+    /// Dequeues a `Queue`
+    pub fn dequeue(ref mut self) -> Option<T> {
+        if self.vec.is_empty() {
+            Option::None
+        } else {
+            Option::Some(self.vec.remove(0))
+        }
+    }
+
+    /// Gets the last element
+    pub fn peek(self) -> Option<T> {
+        self.vec.get(0)
+    }
```

Comment:
This states that it gets the last element but it gets the 0th index which is always the first element

Reply:
That was meaning the last item in the queue, not the underlying `Vec`. Changed to more converntional head/tail terminology.